pkg/importer/importer.go

```diff
@@ -0,0 +1,226 @@
+package importer
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/cortexproject/cortex/pkg/importer/importerpb"
+	"github.com/cortexproject/cortex/pkg/util/services"
+)
+
+type Importer struct {
+	services.Service
+}
+
+func New() *Importer {
+	i := &Importer{}
+	i.Service = services.NewBasicService(i.starting, i.running, i.stopping)
+	return i
+}
+
+func (i *Importer) SampleRPC(ctx context.Context, request *importerpb.SampleRequest) (*importerpb.SampleResponse, error) {
+	//TODO implement me
+	panic("implement me")
+}
+
+func downloadPrometheusBlock(url string) (string, error) {
+	// Create the /tmp directory if it doesn't already exist
+	if _, err := os.Stat("/tmp/tsdb_blocks"); os.IsNotExist(err) {
+		err = os.Mkdir("/tmp/tsdb_blocks", 0755)
+		if err != nil {
+			return "", err
+		}
+	}
```

Comment:
Should we get the `path` of this folder via config? And also this folder should be created "per tenant" and with a random prefix (so we can have multiples requests being processed at the same time?

I see something like: `/data/{tenantId}/{random}`